src/server/ApiMessage.cpp

```diff
@@ -153,7 +153,7 @@ HRESULT _CONSOLE_API_MSG::ReleaseMessageBuffers()
 
     if (State.InputBuffer != nullptr)
     {
-        delete[] State.InputBuffer;
+        delete[] (BYTE**) State.InputBuffer;

```

Comment:
I'd be _really_ wary of changing this, the API code is notoriously fragile. Especially considering you're casting the two `delete[]`s to different types.

Reply:
deleting void* is UB

Reply:
I believe @ebudai is right; in fact, in c++ 17 it's even more dangerous to abuse delete this way because deletion may need to meet specific alignment requirements originally set during new.  I also believe performing a delete on void won't cause destructors to be called.  The callers seems to always use byte, so I believe this to be correct (after it's changed to use C++ style casts).

Quoting from the draft standard:

http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2017/n4659.pdf

> 8.3.5 Delete
> ...
> In the first alternative (delete object), if the static type of the object to be deleted is different from its dynamic
> type, the static type shall be a base class of the dynamic type of the object to be deleted and the static type
> shall have a virtual destructor or the behavior is undefined. In the second alternative (delete array) if the
> dynamic type of the object to be deleted differs from its static type, the behavior is undefined.
> ...
> 82) This implies that an object cannot be deleted using a pointer of type void* because void is not an object type